src/python/openvino_genai/py_openvino_genai.pyi

```diff
@@ -0,0 +1,1648 @@
+        ...
+    def set_chat_template(self, chat_template: str) -> None:
+        """
+        Override a chat_template read from tokenizer_config.json.
+        """
+class UNet2DConditionModel:
+    """
+    UNet2DConditionModel class.
+    """
+    class Config:
+        """
+        This class is used for storing UNet2DConditionModel config.
+        """
+        in_channels: int
+        sample_size: int
+        time_cond_proj_dim: int
+        def __init__(self, config_path: os.PathLike) -> None:
+            ...
+    @typing.overload
+    def __init__(self, root_dir: os.PathLike) -> None:
+        """
+                    UNet2DConditionModel class
+                    root_dir (str): Model root directory.
+        """
+    @typing.overload
+    def __init__(self, root_dir: os.PathLike, device: str, **kwargs) -> None:
+        """
+                    UNet2DConditionModel class
+                    root_dir (str): Model root directory.
+                    device (str): Device on which inference will be done.
+                    kwargs: Device properties.
+        """
+    @typing.overload
+    def __init__(self, model: UNet2DConditionModel) -> None:
+        """
+        UNet2DConditionModel model
+                    UNet2DConditionModel class
+                    model (UNet2DConditionModel): UNet2DConditionModel model
+        """
+    def compile(self, device: str, **kwargs) -> None:
+        """
+                        Compiles the model.
+                        device (str): Device to run the model on (e.g., CPU, GPU).
+                        kwargs: Device properties.
+        """
+    def get_config(self) -> UNet2DConditionModel.Config:
+        ...
+    def infer(self, sample: openvino._pyopenvino.Tensor, timestep: openvino._pyopenvino.Tensor) -> openvino._pyopenvino.Tensor:
+        ...
+    def reshape(self, batch_size: int, height: int, width: int, tokenizer_model_max_length: int) -> UNet2DConditionModel:
+        ...
+    def set_adapters(self, adapters: AdapterConfig | None) -> None:
+        ...
+    def set_hidden_states(self, tensor_name: str, encoder_hidden_states: openvino._pyopenvino.Tensor) -> None:
+        ...
+class VLMPipeline:
+    """
+    This class is used for generation with VLMs
+    """
+    def __init__(self, models_path: os.PathLike, device: str, **kwargs) -> None:
+        """
+        device on which inference will be done
+                    VLMPipeline class constructor.
+                    models_path (str): Path to the folder with exported model files.
+                    device (str): Device to run the model on (e.g., CPU, GPU). Default is 'CPU'.
+                    kwargs: Device properties
+        """
+    def finish_chat(self) -> None:
+        ...
+    @typing.overload
+    def generate(self, prompt: str, images: list[openvino._pyopenvino.Tensor], generation_config: GenerationConfig = None, streamer: typing.Callable[[str], bool] | StreamerBase | None = None, **kwargs) -> typing.Any:
```

Comment:
let's also add overload where user don't have to pass list of tensors, e.g. when I have a single image.

On both C++ and Python levels.

Reply:
Done.